src/cn-flex-form.service.js

```diff
@@ -1704,8 +1703,19 @@ function CNFlexFormService(
   function processSelectDisplay(selectDisplay) {
     const service = this;
     const schema = service.getSchema(selectDisplay.key);
-    const selectField = _.find(selectDisplay.items, 'selectField');
 
+    // Needed for batchform to check recursively
+    let selectField = null;
+    for (let item of selectDisplay.items) {
+      if (item.selectField) {
+        selectField = item;
+      } else if (item.items) {
+        selectField = _.find(item.items, 'selectField');
+      }
+      if (selectField) {
+        break;
+      }
+    }
```

Comment:
Once again this is untested and I might be understanding the logic wrong, but I _think_ this could just be:

```js
let selectField =
  _.find(selectDisplay.items, item =>
    item.selectField || _.find(item.items, 'selectField')
  );
```

Reply:
This doesn't work because it would still return `item` and not the nest `item` in `item.items`

> Iterates over elements of collection, returning the first element predicate returns truthy for. The predicate is invoked with three arguments: (value, index|key, collection).